modules/fedimint-gw-client/src/pay.rs

```diff
@@ -0,0 +1,17 @@
+/// State machine that pays a lightning invoice on behalf of a federation user.
+///
+/// ```mermaid
+/// graph LR
+/// classDef virtual fill:#fff,stroke-dasharray: 5 5
+///
+///    PayInvoice -- validate outgoing contract --> Canceled
+///    PayInvoice -- await preimage offer accepted --> BuyPreimage
+///    PayInvoice -- await preimage offer failure --> Canceled
```

Comment:
What are the two different failure cases here?

Reply:
Removed